cmd/revad/svcs/httpsvcs/ocdavsvc/trashbin.go

```diff
@@ -226,6 +182,166 @@ func (h *TrashbinHandler) listTrashbin(w http.ResponseWriter, r *http.Request, s
+				Prop: []*propertyXML{
+					s.newProp("d:resourcetype", "<d:collection/>"),
+				},
+			},
+			propstatXML{
+				Status: "HTTP/1.1 404 Not Found",
+				Prop: []*propertyXML{
+					s.newProp("oc:trashbin-original-filename", ""),
+					s.newProp("oc:trashbin-original-location", ""),
+					s.newProp("oc:trashbin-delete-datetime", ""),
+					s.newProp("d:getcontentlength", ""),
+				},
+			},
+		},
+	})
+	/*
+		for i := range items {
+			vi := &storageproviderv0alphapb.ResourceInfo{
+				// TODO(jfd) we cannot access version content, this will be a problem when trying to fetch version thumbnails
+				//Opaque
+				Type: storageproviderv0alphapb.ResourceType_RESOURCE_TYPE_FILE,
+				Id: &storageproviderv0alphapb.ResourceId{
+					StorageId: "trashbin", // this is a virtual storage
+					OpaqueId:  path.Join("trash-bin", u.Username, items[i].GetKey()),
+				},
+				//Checksum
+				//Etag: v.ETag,
+				//MimeType
+				Mtime: items[i].DeletionTime,
+				Path:  items[i].Key,
+				//PermissionSet
+				Size:  items[i].Size,
+				Owner: u.Id,
+			}
+			infos = append(infos, vi)
+		}
+	*/
+	for i := range items {
+		res, err := h.itemToPropResponse(ctx, s, pf, items[i])
+		if err != nil {
+			return "", err
+		}
+		responses = append(responses, res)
+	}
+	responsesXML, err := xml.Marshal(&responses)
+	if err != nil {
+		return "", err
+	}
+
+	msg := `<?xml version="1.0" encoding="utf-8"?><d:multistatus xmlns:d="DAV:" `
+	msg += `xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns">`
+	msg += string(responsesXML) + `</d:multistatus>`
+	return msg, nil
+}
+
+// TODO(jfd) render trashbin response
+// <oc:trashbin-original-filename>ownCloud Manual.pdf</oc:trashbin-original-filename>
+// this seems to be relative to the users home ... which is bad: now the client has to build the proper Destination url
+// <oc:trashbin-original-location>ownCloud Manual.pdf</oc:trashbin-original-location>
+// <oc:trashbin-delete-datetime>Thu, 29 Aug 2019 14:06:24 GMT</oc:trashbin-delete-datetime>
+// d:getcontentlength
+// d:resourcetype
+func (h *TrashbinHandler) itemToPropResponse(ctx context.Context, s *svc, pf *propfindXML, item *storageproviderv0alphapb.RecycleItem) (*responseXML, error) {
+
+	baseURI := ctx.Value(ctxKeyBaseURI).(string)
+	t := utils.TSToTime(item.DeletionTime).UTC()
+	dTime := t.Format(time.RFC1123)
+	ref := path.Join(baseURI, item.Key+".d"+dTime)
+	if item.Type == storageproviderv0alphapb.ResourceType_RESOURCE_TYPE_CONTAINER {
+		ref += "/"
+	}
+
+	response := responseXML{
+		Href:     (&url.URL{Path: ref}).EscapedPath(), // url encode response.Href
+		Propstat: []propstatXML{},
+	}
+
+	// when allprops has been requested
+	if pf.Allprop != nil {
+		// return all known properties
+		response.Propstat = append(response.Propstat, propstatXML{
+			Status: "HTTP/1.1 200 OK",
+			Prop:   []*propertyXML{},
+		})
+		response.Propstat[0].Prop = append(response.Propstat[0].Prop, s.newProp("oc:trashbin-original-filename", item.Key))
```

Comment:
I think this should be path also

Reply:
yeah ... we neet to translate a bit what the webdav api wants here: this is from the phoenix.owncloud.com instance whan listing the trashbin
```xml
<d:response>
    <d:href>/remote.php/dav/trash-bin/demo/197</d:href>
    <d:propstat>
        <d:prop>
            <oc:trashbin-original-filename>.hidden</oc:trashbin-original-filename>
            <oc:trashbin-original-location>First/Second/Third/.hidden</oc:trashbin-original-location>
            <oc:trashbin-delete-datetime>Wed, 05 Jun 2019 10:10:14 GMT</oc:trashbin-delete-datetime>
            <d:getcontentlength>0</d:getcontentlength>
            <d:resourcetype/>
        </d:prop>
        <d:status>HTTP/1.1 200 OK</d:status>
    </d:propstat>
</d:response>
```
so yeah ... `oc:trashbin-original-filename` is completely redundant ... will be gzipped away ... so we can just use the basename of `oc:trashbin-original-location` which is the CS3 Path.

I'll update the PR ...